alphagov/paas-cf
# INVALID [#132909443] Upgrade stemcell 3263 and migrate to garden-runc

[#132909443 Respond to CVE-2016-5195 kernel vulnerability (AKA upgrade to stemcell 3263 and garden-runc)](https://www.pivotaltracker.com/story/show/132909443)

What?
-----

[A vulnerability that allows escalation of privileges CVE-2016-5195 (aka dirtycow)](http://dirtycow.ninja/) in the linux kernel has been reported recently. Our systems are affected, but specially worrying are the cells and the containers running on it.

Despite being reticent, Cloud Foundry community included a [patched version in a older series with 3.3](https://cloudfoundry.slack.com/archives/bosh/p1477113219004741) on 3262.22. That has been done in the PR #560

But in this PR we want to upgrade to stemcell 3263 and migrate to garden-linux:

 * We learn that the stemcell 3263 includes the patch, but it runs the kernel 4.4. But this kernel can cause problems with garden-linux ~~and Cloud Foundry community will not patch older stemcells with the 3.3 kernel~~ , . 
 * Also, garden-linux is not supported anymore.

This change represents some risk in the cells running the garden containers. To allow a quick manual rollback if required, we keep in the manifest the definitions of the garden-linux release and stemcell 3262. We need to add a dummy job definition `cell_3_3` to avoid the stemcell and release being deleted by the bosh-clean job (see #558)

Dependencies
------------

This code is based on the upgrade of CF 245 in #557 and that should be merge first and this branch rebased.

How to test?
------------

Deploy this release. All test should pass OK.

Note: As it is a stemcell change, all VMs will be reprovision.

Who?
----

Anyone but @keymon
